src/Core/Document/ErrorSource.php

```diff
@@ -69,11 +75,13 @@ public static function fromArray(array $source): self
      *
      * @param string|null $pointer
      * @param string|null $parameter
+     * @param string|null $header
      */
-    public function __construct(?string $pointer = null, ?string $parameter = null)
+    public function __construct(string $pointer = null, string $parameter = null, string $header = null)
```

Comment:
Note: Implicit nullable parameters (string $parameter = null) have been deprecated in PHP 8.4

Reply:
Given that it was like this before, and all tests pass, do you want me to do that? Or is clean PHP 8.4 not a strict goal?

Reply:
Sorry, bit confused! It was `?string` before, you've changed it to `string` which is an implicit nullable, which is deprecated. It needs to be an explicit nullable, which is `?string $pointer = null` - which is what it was. 

I removed all the PHP 8.4 deprecation notices back in November; the change you've made will reintroduce it.

Reply:
I see, I originally made my change on an old version, then messed that up when I copied them to HEAD. I will fix.

Why not check for that in the github actions? I use https://github.com/phan/phan/ for that in my projects.

Reply:
Thanks. I'll need to check why the action didn't fail as it's meant to fail for deprecation notices.

Reply:
Ah, it didn't fail because there isn't test coverage of this class. Need to sort that out!